cishouseholds/merge.py

```diff
@@ -40,10 +47,22 @@ def union_multiple_tables(tables: List[DataFrame]):
         list of objects representing the respective input tables
     """
     merged_df = tables[0]
-
     for i, table_n in enumerate(tables[1:]):
-        merged_df, dfn = prepare_for_union(merged_df, table_n)
-        merged_df = merged_df.unionByName(dfn)
+        merged_df, dfn = prepare_for_union(merged_df, table_n, combine_order=reorder_columns)
+        config = get_config()
+        if config != {}:
```

Comment:
Please don't use this assumption to determine whether the code is being run in test or not. You should separate testable logic from the table writing to achieve this.